mobile/src/main/java/org/openhab/habdroid/ui/WidgetAdapter.kt

```diff
@@ -780,6 +785,46 @@ class WidgetAdapter(
         }
     }
 
+    class ButtongridViewHolder internal constructor(private val initData: ViewHolderInitData) :
+        LabeledItemBaseViewHolder(initData, R.layout.widgetlist_buttongriditem), View.OnClickListener {
+        private val table: GridLayout = itemView.findViewById(R.id.widget_content)
+
+        override fun bind(widget: Widget) {
+            super.bind(widget)
+            val mappings = widget.mappings.filter { it.column != 0 && it.row != 0 }
+            table.rowCount = mappings.maxOfOrNull { it.row } ?: 0
+            table.columnCount = mappings.maxOfOrNull { it.column } ?: 0
+            (0 until table.rowCount).forEach { row ->
+                (0 until table.columnCount).forEach { column ->
+                    val buttonView = initData.inflater.inflate(R.layout.widgetlist_sectionswitchitem_button, null) as MaterialButton
```

Comment:
You need similar logic as for section switch here (remove old views, ideally keeping them in a pool for reuse), otherwise you'll duplicate views when the holder is reused.

Reply:
https://github.com/openhab/openhab-android/pull/3546/commits/9c8cae438003c27927947f51a5965252591d6bc5